### ISSUE TITLE
lr_scheduler seems to be reset if training is resumed

### ISSUE THREAD

**Issue by julian90** [awslabs/sockeye#366]
Hi,

if training is resumed the learning rate scheduler seems to be reset.

This seems to be the case because the class OptimizerConfig contains the learning rate scheduler both directly in lr_scheduler and also in the dict params, and the module.init_optimizer function relies on the dict OptimizerConfig.params but not on OptimizerConfig.lr_scheduler.

But here:
https://github.com/awslabs/sockeye/blob/da518f1c6e90e08e2a1f8d2888608121f75b3d67/sockeye/training.py#L989-L993
only the lr_scheduler is updated but not the params dict.

A quick fix might be adding

`self.optimizer_config.params['lr_scheduler'] = self.optimizer_config.lr_scheduler`

before

https://github.com/awslabs/sockeye/blob/da518f1c6e90e08e2a1f8d2888608121f75b3d67/sockeye/training.py#L993

But overall it looks dangerous to have the learning rate scheduler both in lr_scheduler and in the dict params within the class OptimizerConfig.

**Comment by fhieber** [comment]
Hi @julian90, thanks for submitting this issue! This is clearly a bug and I agree about the double reference to the scheduler instance in OptimizerConfig.
Does this [change](https://github.com/awslabs/sockeye/commit/778bdcf0a839e44fdf45c323e667bb900a52fef0) fixes the problem for you?
